web/app/services/auth.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import {HttpEvent, HttpHandler, HttpInterceptor, HttpRequest} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {LocalStorageKeys} from '../common/constants';
+
+/**
+ * Intercepts the outgoing HTTP requests and attaches the stored JSON web token
+ * to the HTTP header as an Auth Bearer. This saves us from having to attach the
+ * auth token on every request me make manually.
+ */
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  intercept(req: HttpRequest<any>, next: HttpHandler):
+      Observable<HttpEvent<any>> {
+    const token = localStorage.getItem(LocalStorageKeys.AUTH_TOKEN);
+
+    // Attach auth token to request if it exists. Otherwise, handle it normally.
+    if (token) {
+      const authorizedRequest = req.clone(
+          {headers: req.headers.set('Authorization', `Bearer ${token}`)});
```

Comment:
might receive always a `403` when the token has expired, maybe this `Interceptor` should try to fetch a new token?

Reply:
eventually i'd like to store the expiry of the token to determine if the user is still logged in. If the token is expired we should block the requests, until they log in.

Reply:
@nakhbari let's create an issue to track this idea 👍

Reply:
it's one of the subtasks in #752